### PR TITLE
fix: Make sure that rollback hook is triggered on all version backends

### PR DESCRIPTION
**Pull request by juliusknorr**
## Summary

Alternative to https://github.com/nextcloud/groupfolders/pull/2260 to apply to any version backend.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by juliushaertl** [comment]
Will have a look into the failing tests...

File: apps/files_versions/lib/Versions/VersionManager.php [hunk 1]
```diff
@@ -94,7 +94,13 @@ public function createVersion(IUser $user, FileInfo $file) {
 
 	public function rollback(IVersion $version) {
 		$backend = $version->getBackend();
-		return $backend->rollback($version);
+		$result = $backend->rollback($version);
+		\OC_Hook::emit('\OCP\Versions', 'rollback', [
+			'path' => \OC\Files\Filesystem::getView()->getRelativePath($version->getSourceFile()->getPath()),
+			'revision' => $version->getRevisionId(),
+			'node' => $version->getSourceFile(),
+		]);
```
Comment (artonge) [inline apps/files_versions/lib/Versions/VersionManager.php:102]:
```suggestion
if ($result) {
		\OC_Hook::emit('\OCP\Versions', 'rollback', [
			'path' => \OC\Files\Filesystem::getView()->getRelativePath($version->getSourceFile()->getPath()),
			'revision' => $version->getRevisionId(),
			'node' => $version->getSourceFile(),
		]);
}
```

**Review by artonge — APPROVED** [review]

**Comment by juliushaertl** [comment]
Thanks a lot @icewind1991 💙

**Comment by juliushaertl** [comment]
@icewind1991 Kindly also asking for a review :)

**Comment by AndyXheli** [comment]
Hello, Is there any updates on this. We created a ticket with onlyoffice regarding this since we're enterprise user with them. Looks like there was a sugestion fix via https://github.com/nextcloud/groupfolders/pull/2260 Can Nextcloud Devs look into this as our users with onlyofice are effected by this. Thank you!

**Comment by juliushaertl** [comment]
@AndyXheli Please make sure to reach out through support if this is affecting enterprise customers ;)

**Comment by juliushaertl** [comment]
Sorry accidentally cleared the review @artonge Nothing changed since back then.

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**